tests/UnifiedSpecTests/Context.php

```diff
@@ -256,7 +257,15 @@ private function createClient(string $id, stdClass $o)
         static $i = 0;
         $driverOptions = isset($observeEvents) ? ['i' => $i++] : [];
 
-        $this->entityMap->set($id, new Client($uri, $uriOptions, $driverOptions));
+        if ($serverApi !== null) {
+            $driverOptions['serverApi'] = new ServerApi(
+                $serverApi->version,
+                $serverApi->strict ?? null,
+                $serverApi->deprecationErrors ?? null
+            );
+        }
+
+        $this->entityMap->set($id, UnifiedSpecTest::createTestClient($uri, $uriOptions, $driverOptions));
```

Comment:
If the test suite is run against a server with `requireApiVersion` enabled, all client entities must have the `serverApi` options set even if it isn't expressed in the test. The unified test format currently doesn't allow for specifying `serverApi` as `null` to indicate that no server API Version should be used, but this could be added if it's relevant.

As it is, the logic in `createTestClient` does not overwrite a `serverApi` option, so tests can still specify their own versions.

Reply:
> The unified test format currently doesn't allow for specifying serverApi as null to indicate that no server API Version should be used, but this could be added if it's relevant.

We can consider that outside of the PR. I suppose it depends on whether enough folks want a unified test for the error case. Personally, I'm fine with handling that in prose.